### PR TITLE
Exercise: Bean counting

### PR DESCRIPTION
**Pull request by artem-konovalov-personal**
Closed [#205](https://github.com/nesterone/sf-eng-init-gl-spring-2016/issues/205)

### CODE REVIEW

File: 01-syntax/02-functions-variables-scope-and-memory/bean-counting/countBs.js [hunk 1]
```diff
@@ -4,7 +4,18 @@
...
+  for (charCount = 0; charCount < text.length; charCount++) {
+    if (text.charAt(charCount) === 'B') {
+      numberOfCharacters += 1;
+    }
+  }
+
+  return numberOfCharacters;
+}
 
 console.log(countBs('BBC'));
```
Comment (nesterone) [inline 01-syntax/02-functions-variables-scope-and-memory/bean-counting/countBs.js:11]: Please, extend with use cases, when consumer of your function run `countBs();`

File: 01-syntax/02-functions-variables-scope-and-memory/bean-counting/countChar.js [hunk 2]
```diff
@@ -4,7 +4,18 @@
...
+  for (charCount = 0; charCount < text.length; charCount++) {
+    if (text.charAt(charCount) === character) {
+      numberOfCharacters += 1;
+    }
+  }
+
+  return numberOfCharacters;
+}
 
 console.log(countChar('kakkerlak', 'k'));
```
Comment (nesterone) [inline 01-syntax/02-functions-variables-scope-and-memory/bean-counting/countChar.js:25]: The same here, try to think about edge cases, on how function could be invoked by other developer

**Comment by artem-konovalov-personal** [comment]
I have added few checks for functions to the both files  `countBs.js` and countChar.js``

File: 01-syntax/02-functions-variables-scope-and-memory/bean-counting/countBs.js [hunk 3]
```diff
@@ -4,7 +4,48 @@
  */
 /* global countBs */
 
-// → Your code here.
+function countBs(text) {
+  var charCount;
+  var numberOfCharacters = 0;
+
+  if (! text || text.length === 0 ||
```
Comment (nesterone) [inline 01-syntax/02-functions-variables-scope-and-memory/bean-counting/countBs.js]: Any ideas on how to reuse `countChar` here ?

DRY!

**Comment by artem-konovalov-personal** [comment]
I have changed the code's structure. Now It doesn't contain the same code

**Pull request merged**